libs/providers/flagd/src/lib/service/http/service.ts

```diff
@@ -0,0 +1,165 @@
+          errorCode: ErrorCode.PARSE_ERROR,
+          variant: "default_value",
+        }
+      } catch (err: unknown) {
+        return {
+          reason: StandardResolutionReasons.ERROR,
+          errorCode: GetErrorCode(err),
+          variant: 'default_value',
+          value: defaultValue,
+        }
+      }
+    }
+
+    async ResolveString(flagKey: string, defaultValue: string, context: EvaluationContext): Promise<ResolutionDetails<string>> {
+      try {
+        const res = await axios.post<ResolveStringResponse>(
+          `${this.url}/flags/${flagKey}/resolve/string`,
+          context
+        );
+        if (CheckResponse(res, 'string')) {
+          return {
+            value: res.data.value,
+            reason: res.data.reason,
+            variant: res.data.variant,
+          };
+        }
+        return {
+          value: defaultValue,
+          reason: StandardResolutionReasons.ERROR,
+          errorCode: ErrorCode.PARSE_ERROR,
+          variant: "default_value",
+        }
+      } catch (err: unknown) {
+        return {
+          reason: StandardResolutionReasons.ERROR,
+          errorCode: GetErrorCode(err),
+          variant: 'default_value',
+          value: defaultValue,
+        }
+      }
+    }
+
+    async ResolveObject<T extends object>(flagKey: string, defaultValue: T, context: EvaluationContext): Promise<ResolutionDetails<T>> {
+      try {
+        const res = await axios.post<ResolveObjectResponse>(
+          `${this.url}/flags/${flagKey}/resolve/object`,
+          context
+        );
+        if (CheckResponse(res, 'object')) {
+          return {
+            value: res.data.value as T,
+            reason: res.data.reason,
+            variant: res.data.variant,
+          };
+        }
+        return {
+          value: defaultValue,
+          reason: StandardResolutionReasons.ERROR,
+          errorCode: ErrorCode.PARSE_ERROR,
+          variant: "default_value",
+        }
+      } catch (err: unknown) {
+        return {
+          reason: StandardResolutionReasons.ERROR,
+          errorCode: GetErrorCode(err),
+          variant: 'default_value',
+          value: defaultValue,
+        }
+      }
+    }
+}
+
+function GetErrorCode(err: unknown): string {
+  const code = (err as Partial<AxiosError>)?.response?.status
+  let res: string = StandardResolutionReasons.UNKNOWN
+  if (code != undefined) {
+    if (code == 404) {
+      res = ErrorCode.FLAG_NOT_FOUND
+    } else if (code == 400) {
+      res = ErrorCode.TYPE_MISMATCH
+    }
+  }
+  return res
+}
+
+function CheckResponse(res: AxiosResponse<ResolutionDetails<unknown>> | AxiosResponse<ResolveObjectResponse>, valueType: string): boolean {
+    if ((res.data)
+    && (res.data.value && typeof res.data.value === valueType)
+    && (res.data.variant && typeof res.data.variant === "string")
+    && (res.data.reason && typeof res.data.reason === "string")
+    ) {
+        return true
+    }
```

Comment:
I think you can simplify this with [optional chaining](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html)

Reply:
ive trimmed it right down - and added a small unit test for it, happy to use optional chaining, but no longer checks for res.data.value first - if its undefined its not a string so returns false etc